src/Sarif/Writers/SarifOneZeroZeroLogger.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.CodeAnalysis.Sarif.Readers;
+using Microsoft.CodeAnalysis.Sarif.VersionOne;
+using Microsoft.CodeAnalysis.Sarif.Visitors;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace Microsoft.CodeAnalysis.Sarif.Writers
+{
+    /// <summary>
+    /// A SARIF logger that works by generating a SARIF v2 log file which is transformed into 
+    /// SARIF v1 when the instance is disposed. The file location used to produced the preliminary
+    /// v2 content is overwritten in place to produce the transformed v1 file.
+    /// </summary>
+    public class SarifOneZeroZeroLogger : SarifLogger
+    {
+        private string _outputFilePath;
+
+        public SarifOneZeroZeroLogger(
+            string outputFilePath,
+            LoggingOptions loggingOptions = SarifLogger.DefaultLoggingOptions,
+            OptionallyEmittedData dataToInsert = OptionallyEmittedData.None,
+            Tool tool = null,
+            Run run = null,
+            IEnumerable<string> analysisTargets = null,
+            IEnumerable<string> invocationTokensToRedact = null,
+            IEnumerable<string> invocationPropertiesToLog = null,
+            string defaultFileEncoding = null)
+            : base(new StreamWriter(new FileStream(outputFilePath, FileMode.Create, FileAccess.Write, FileShare.None)),
+                  loggingOptions,
+                  dataToInsert,
+                  tool,
+                  run,
+                  invocationTokensToRedact,
+                  invocationPropertiesToLog)
+        {
+            _outputFilePath = outputFilePath;
+        }
+
+        public override void Dispose()
+        {
+            base.Dispose();
+
+            JsonSerializerSettings v2Settings = new JsonSerializerSettings()
```

Comment:
Shouldn't `base.Dispose` happen last? Conceivably the derived class `Dispose` method might want to use a disposable resource held by the base class, and now it's gone.

Reply:
base.Dispose() finishes writing the original SARIF log, which I think is then needed here--this seems correct to me, at least as far as the base class is implemented.

---
In reply to: [216135766](https://github.com/Microsoft/sarif-sdk/pull/1077#discussion_r216135766) [](ancestors = 216135766)